File: client/src/app/models/drink.ts
export enum DrinkType {
    BEER = "Beer",
    WINE = "Wine",
    COCKTAIL = "Cocktail",
    LIQUEUR = "Liqueur",
    COFFEE = "Coffee"
}

export class Drink {
    constructor(private id: Number,
                private name: string,
                private details: string,
                private type: DrinkType,
                private price: number,
                private imagePath: string) {}

    public getId(): Number {
        return this.id;
    }

    public getType(): DrinkType {
        return this.type;
    }

    public getName(): string {
        return this.name;
    }

    public getDetails(): string {
        return this.details;
    }
}

File: src/main/java/com/example/bootrestaurant/Drink/Drink.java
package com.example.bootrestaurant.Drink;

import javax.persistence.*;

@Entity
public class Drink {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column
    private String details;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private DrinkType drinkType;

    @Column(nullable = false)
    private Double price;

    @Column
    private String imagePath;

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public DrinkType getDrinkType() {
        return drinkType;
    }

    public void setDrinkType(DrinkType type) {
        this.drinkType = type;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }
}
